src/factories/BaseFactory.h

```diff
@@ -72,13 +73,15 @@ class BaseExporter {
 class BaseFactory {
 public:
     virtual std::optional<std::shared_ptr<IParsedData>> parse(std::vector<uint8_t>& buffer, YAML::Node& data) = 0;
+    virtual std::optional<std::shared_ptr<IParsedData>> parse_modding(std::vector<uint8_t>& buffer, YAML::Node& data) = 0;
     std::optional<std::shared_ptr<BaseExporter>> GetExporter(ExportType type) {
         auto exporters = this->GetExporters();
         if (exporters.find(type) != exporters.end()) {
             return exporters[type];
         }
         return std::nullopt;
     }
+    virtual bool SupportModdedAssets() = 0;
```

Comment:
could this have a default of returning false and have this be overrided by factories that can end up with true? or is it better structure to leave as pure virtual